enterprise/internal/github_apps/store/store.go

```diff
@@ -90,16 +96,30 @@ func (s *gitHubAppsStore) Create(ctx context.Context, app *ghtypes.GitHubApp) (i
 	if err != nil {
 		return -1, err
 	}
+
+	baseURL, err := url.Parse(app.BaseURL)
+	baseURL = extsvc.NormalizeBaseURL(baseURL)
 	domain := app.Domain
 	if domain == "" {
 		domain = types.ReposDomain
 	}
 
+	// We enforce that GitHub Apps created in the "batches" domain are for unique instance URLs.
+	if domain == types.BatchesDomain {
+		existingGHApp, err := s.GetByDomain(ctx, &domain, baseURL.String())
+		if err != nil {
+			fmt.Println(err)
+		}
```

Comment:
This error likely means we just didn't find an existing GH App, right? We could probably double-check, kinda like in your test, and then if it's some other error make sure we're not just eating it:

```suggestion
		// An error is expected if no existing app was found, but we double check that
		// we didn't get a different, unrelated error
		if err != nil && !strings.Contains(err.Error(), "no app exists matching criteria") {
			return -1, errors.Wrap(err, "checking for existing batches app")
		}
```